src/runtime/runtime.cs

```diff
@@ -393,6 +393,10 @@ internal static void Shutdown(ShutdownMode mode)
                 {
                     Py_Finalize();
                 }
+                else
+                {
+                    PyGILState_Release(state);
```

Comment:
This change seems unrelated, why did you add this?

Reply:
Without this `Py_Finalize` called when Python finally shuts down hangs because it can't acquire GIL.